File: Python/백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
a,b = map(int,input().split())
gm = 1
m = 2
while(m<=a and m<=b):
    if(a%m==0 and b%m==0):
        a = a//m
        b = b//m
        gm*=m
        m = 2
    else:
        m+=1
print(gm, gm*a*b)

File: C#/백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.cs
using System;

namespace q{
    class w{
        static int gcd(int a,int b){
            if(b==0){
                return a; 
            }
            return gcd(b,a%b);
        }
        static void Main(){
            int[] aa = Array.ConvertAll(Console.ReadLine().Split(),int.Parse);
            if(aa[1]>aa[0]){
                int temp = aa[1];
                aa[1] = aa[0];
                aa[0] = temp;
            }
            int d = gcd(aa[0],aa[1]);
            int m = (aa[0]*aa[1])/d;
            Console.WriteLine($"{d}\n{m}");
        }
    }
}